server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PlainHighlighter.java

```diff
@@ -218,6 +218,9 @@ private static int findGoodEndForNoHighlightExcerpt(int noMatchSize, Analyzer an
                 // Can't split on term boundaries without offsets
                 return -1;
             }
+            if (contents.length() <= noMatchSize) {
```

Comment:
Should we move this to like 216, even before analyzing `contents` string?

Reply:
Was asking myself the same, but I think the behaviour would not be consistent which could cause confusion - when no offsets are provided we would some return highlights for some document fields, but not for others and the reason why that happens might not be apparent.
If that's not actually a problem I am okay to change it.

Reply:
Thinking more about it, you are right. It is better to have the consistent behaviour. Let's keep your changes as they are.